Add a blog post about fast super property access (#528)
v8/v8.dev: src/blog/fast-super.md

### Prototypal inheritance and `super`

Let's start from the basics - what does super property access even mean?

```javascript
class A { }
A.prototype.x = 100;

class B extends A {
  m() {
    return super.x;
  }
}
const b = new B();
b.m();
```

Now `A` is the super class of `B` and  `b.m()` returns `100` as you'd expect.

![Class inheritance diagram](/_img/fast-super/inheritance-1.svg)

The reality of [JavaScript's prototypal inheritance](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain) is more complicated:

![Prototypal inheritance diagram](/_img/fast-super/inheritance-2.svg)

We need to distinguish carefully between the `__proto__` and `prototype` properties - they don't mean the same thing! To make it more confusing, the object `b.__proto__` is often referred to as "`b`'s prototype".

> `b.__proto__` is the object from which `b` inherits properties. `B.prototype` is the object which will be the `__proto__` of objects created with `new B()`, such as `b`.

Reviewer: I think this minor note would help to clarify things a bit:

```suggestion
`b.__proto__` is the object from which `b` inherits properties. `B.prototype` is the object which will be the `__proto__` of objects created with `new B()`, that is `b.__proto__ === B.prototype`.
```

> `b` has `B.prototype` and `A.prototype` in its prototype chain (that is, the chain of `__proto__` properties), so it can access all properties defined in those objects. The method `m` is a property of `B.prototype`, `B.prototype.m`, and this is why `b.m()` works.

Reviewer: And, correspondingly, to visualise this next paragraph too:

```suggestion
In turn, `B.prototype` itself has its own `__proto__` property that equals to `A.prototype`. Together, this forms what's called a prototype chain:

```
b.__proto__ ->
B.prototype ->
B.prototype.__proto__ ->
A.prototype ->
[...]
```

Through this chain, `b` can access all properties defined in any of those objects. The method `m` is a property of `B.prototype` — `B.prototype.m` — and this is why `b.m()` works.
```
